File: sdks/dotnet/sulfone-helium/Api/Core/AnswerReq.cs
using System.Text.Json.Serialization;
using Swashbuckle.AspNetCore.Annotations;

namespace sulfone_helium.Api.Core;

[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
[SwaggerDiscriminator("type")]
[JsonDerivedType(typeof(StringAnswerReq), typeDiscriminator: "string")]
[SwaggerSubType(typeof(StringAnswerReq), DiscriminatorValue = "string")]
[JsonDerivedType(typeof(BoolAnswerReq), typeDiscriminator: "boolean")]
[SwaggerSubType(typeof(BoolAnswerReq), DiscriminatorValue = "boolean")]
[JsonDerivedType(typeof(StringArrayAnswerReq), typeDiscriminator: "str_array")]
[SwaggerSubType(typeof(StringArrayAnswerReq), DiscriminatorValue = "str_array")]
public abstract class AnswerReq
{
}

public class StringArrayAnswerReq : AnswerReq
{
    public required string[] Answer { get; set; }
}

public class StringAnswerReq : AnswerReq
{
    public required string Answer { get; set; }
}

public class BoolAnswerReq : AnswerReq
{
    public required bool Answer { get; set; }
}

File: sdks/node/src/api/core/answer_req.ts
interface StringArrayAnswerReq {
  readonly answer: string[];
}

interface StringAnswerReq {
  readonly answer: string;
}

interface BoolAnswerReq {
  readonly answer: boolean;
}

type AnswerReq = StringArrayAnswerReq | StringAnswerReq | BoolAnswerReq;

function isStringAnswerReq(a: AnswerReq): a is StringAnswerReq {
  return typeof a.answer === "string";
}

function isStringArrayAnswerReq(a: AnswerReq): a is StringArrayAnswerReq {
  return Array.isArray(a.answer);
}

function isBoolAnswerReq(a: AnswerReq): a is BoolAnswerReq {
  return typeof a.answer === "boolean";
}

export { isStringAnswerReq, isStringArrayAnswerReq, isBoolAnswerReq };

export type { AnswerReq, StringAnswerReq, StringArrayAnswerReq, BoolAnswerReq };
